CMakeLists.txt

```diff
@@ -1601,15 +1617,28 @@ target_include_directories(libnetdata BEFORE PUBLIC ${LIBUV_INCLUDE_DIRS})
 target_compile_definitions(libnetdata PUBLIC ${LIBUV_CFLAGS_OTHER})
 target_link_libraries(libnetdata PUBLIC ${LIBUV_LDFLAGS})
 
-# crypto
-target_include_directories(libnetdata BEFORE PUBLIC ${CRYPTO_INCLUDE_DIRS})
-target_compile_options(libnetdata PUBLIC ${CRYPTO_CFLAGS_OTHER})
-target_link_libraries(libnetdata PUBLIC ${CRYPTO_LDFLAGS})
+if (ENABLE_OPENSSL)
+        message(STATUS "Compiling Netdata with OpenSSL")
+        # crypto
+        target_include_directories(libnetdata BEFORE PUBLIC ${CRYPTO_INCLUDE_DIRS})
+        target_compile_options(libnetdata PUBLIC ${CRYPTO_CFLAGS_OTHER})
+        target_link_libraries(libnetdata PUBLIC ${CRYPTO_LDFLAGS})
+
+        # openssl
+        target_include_directories(libnetdata BEFORE PUBLIC ${OPENSSL_INCLUDE_DIRS})
+        target_compile_options(libnetdata PUBLIC ${OPENSSL_CFLAGS_OTHER})
+        target_link_libraries(libnetdata PUBLIC ${OPENSSL_LDFLAGS})
+endif()
 
-# openssl
-target_include_directories(libnetdata BEFORE PUBLIC ${OPENSSL_INCLUDE_DIRS})
-target_compile_options(libnetdata PUBLIC ${OPENSSL_CFLAGS_OTHER})
-target_link_libraries(libnetdata PUBLIC ${OPENSSL_LDFLAGS})
+if (ENABLE_WOLFSSL)
+        message(STATUS "Compiling Netdata with WolfSSL")
+
+        target_include_directories(libnetdata BEFORE PUBLIC ${WOLFSSL_INCLUDE_DIRS})
+        target_compile_options(libnetdata PUBLIC ${WOLFSSL_CFLAGS_OTHER}
+                               -DENABLE_WOLFSSL
+                               -DWOLFSSL_BASE64_ENCODE)
```

Comment:
These should be handled via `packaging/cmake/conf.h.in`.

Reply:
Done!